guacamole/src/main/webapp/app/rest/services/requestService.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Service for converting $http promises that pass the entire response into
+ * promises that pass only the data from that response.
+ */
+angular.module('rest').factory('requestService', ['$q', '$http', 'Error',
+        function requestService($q, $http, Error) {
+
+    function wrappedHttpCall(object) {
+        
+        var deferred = $q.defer();
+        
+        $http(object)
```

Comment:
Perhaps it would be cleaner to leverage the `return` / `throw` magic provided by promises? Something like:

```js
return $http(object).then(
    function success(request) { return request.data; },
    function failure(request) { throw new Error(request.data); }
);
```

would remove one layer from the promise onion and the need for manual use of `$q`.

Reply:
Good call. Updated and it looks much cleaner that way.